### PR TITLE
Task 189 implement cucumber steps for file input element

### PR DESCRIPTION
**Pull request by v5v5**
Implemented 'Feature: File Input element functionality' with steps:
- When I upload file "/src/test/resources/general.xml" by "Avatar" file input element
- When I try to upload file "/jdi-light-bdd-tests/src/test/resources/general.xml" by "Disabled File Input" file input element
- Then "Avatar" file input element label is "Profile picture:"
- Then "Avatar" file input element label contains "picture"
- Then "Avatar" file input element text is "C:\fakepath\general.xml"
- Then "Avatar" file input element text contains "general.xml"
- Then "Avatar" file input element value is "C:\fakepath\general.xml"
- Then "Avatar" file input element value contains "general.xml"

in scenarios:
- Scenario: Upload file by enabled file input element
- Scenario: Upload file by disabled file input element

### CODE REVIEW

File: jdi-light-bdd-tests/src/test/java/cucmberTests/stepdefs/baseCucumberScenario.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import static io.github.com.StaticSite.homePage;
+import static io.github.com.entities.Users.DEFAULT_USER;
+import static io.github.com.pages.Header.loginForm;
+import static io.github.com.pages.Header.userIcon;
+
+/**
+ * Created by Dmitry_Lebedev1 on 1/13/2016.
+ */
+public class baseCucumberScenario {
+
```
Comment (RomanIovlev) [inline jdi-light-bdd-tests/src/test/java/cucmberTests/stepdefs/baseCucumberScenario.java]: remove this block of code

File: jdi-light-bdd-tests/src/test/resources/features/FileInput.feature [hunk 2]
```diff
@@ -0,0 +1,23 @@
+Feature: File Input element functionality
+
+  Scenario: Upload file by enabled file input element
```
Comment (RomanIovlev) [inline jdi-light-bdd-tests/src/test/resources/features/FileInput.feature]: split to smaller tests

File: jdi-light-bdd-tests/src/test/resources/features/FileInput.feature [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    Given I open "Html5 Page"
+    Then "Avatar" label text equals to "Profile picture:"
+    And "Avatar" label text contains "picture"
+    And "Avatar" is enabled
```
Comment (RomanIovlev) [inline jdi-light-bdd-tests/src/test/resources/features/FileInput.feature]: remove this basic validations
And add step with all basic validations

File: jdi-light-bdd-tests/src/test/resources/features/FileInput.feature [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    And "Avatar" is displayed
+    And "Avatar" attribute "id" equals to "avatar"
+    When I upload file "/src/test/resources/general.xml" by "Avatar" file input element
+    Then "Avatar" text equals to "C:\fakepath\general.xml"
+    And "Avatar" text contains "general.xml"
+    And "Avatar" file input element value equals to "C:\fakepath\general.xml"
```
Comment (RomanIovlev) [inline jdi-light-bdd-tests/src/test/resources/features/FileInput.feature]: replace with 
    And "Avatar" text equals to "C:\fakepath\general.xml"
    And "Avatar" text contains "general.xml"

File: jdi-light-bdd-tests/src/test/resources/features/FileInput.feature [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    And "Avatar" file input element value contains "general.xml"
+
+  Scenario: Upload file by disabled file input element
```
Comment (RomanIovlev) [inline jdi-light-bdd-tests/src/test/resources/features/FileInput.feature]: Test fails

File: jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import static com.epam.jdi.tools.PathUtils.mergePath;
+import static org.junit.Assert.assertEquals;
+
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.MatcherAssert;
+
+public class FileInputSteps {
+    @When("^I upload file \"([^\"]*)\" by \"([^\"]*)\" file input element$")
+    public void iUploadFileByFileInputElement(String pathToFile, String elementName) {
+        FileInput fileInput = (FileInput) Utils.getUI(elementName);
```
Comment (RomanIovlev) [inline jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java]: use getUI(elementName, FileInpu.class); instead

File: jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        fileInput.uploadFile(mergePath(PROJECT_PATH, pathToFile));
+    }
+
+    @When("^I try to upload file \"([^\"]*)\" by \"([^\"]*)\" file input element$")
+    public void iTryToUploadFileByFileInputElement(String pathToFile, String elementName) {
```
Comment (RomanIovlev) [inline jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java]: remove this step from element steps and move in UserSteps in bdd tests

File: jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        try {
+            fileInput.uploadFile(mergePath(PROJECT_PATH, pathToFile));
+        } catch (Exception e) {
+            assertEquals(
+                    String.format("\r\nCan't do uploadFile for disabled element '%s'", elementName),e.getMessage());
+        }
+    }
+
+    @And("^\"([^\"]*)\" file input element value equals to \"([^\"]*)\"$")
+    public void fileInputElementValueEqualsTo(String elementName, String value) {
```
Comment (RomanIovlev) [inline jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java]: remove this step
equal to    And "Avatar" text equals to "C:\fakepath\general.xml"

File: jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        FileInput fileInput = (FileInput) Utils.getUI(elementName);
+        assertEquals(fileInput.value(), value);
+    }
+
+    @And("^\"([^\"]*)\" file input element value contains \"([^\"]*)\"$")
```
Comment (RomanIovlev) [inline jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java]: remove this step
equal to        And "Avatar" text contains "general.xml"

File: jdi-light-bdd-tests/src/test/resources/features/FileInput.feature (hunk 2 above)
Comment (v5v5) [reply jdi-light-bdd-tests/src/test/resources/features/FileInput.feature]: fixed

File: jdi-light-bdd-tests/src/test/resources/features/FileInput.feature (hunk 3 above)
Comment (v5v5) [reply jdi-light-bdd-tests/src/test/resources/features/FileInput.feature]: fixed

File: jdi-light-bdd-tests/src/test/resources/features/FileInput.feature (hunk 5 above)
Comment (v5v5) [reply jdi-light-bdd-tests/src/test/resources/features/FileInput.feature]: Localy test passed

File: jdi-light-bdd-tests/src/test/java/cucmberTests/stepdefs/baseCucumberScenario.java (hunk 1 above)
Comment (v5v5) [reply jdi-light-bdd-tests/src/test/java/cucmberTests/stepdefs/baseCucumberScenario.java]: fixed

File: jdi-light-bdd-tests/src/test/resources/features/FileInput.feature (hunk 4 above)
Comment (v5v5) [reply jdi-light-bdd-tests/src/test/resources/features/FileInput.feature]: fixed

File: jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java (hunk 6 above)
Comment (v5v5) [reply jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java]: fixed

File: jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java (hunk 8 above)
Comment (v5v5) [reply jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java]: fixed

File: jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java (hunk 9 above)
Comment (v5v5) [reply jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java]: fixed

File: jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java (hunk 7 above)
Comment (v5v5) [reply jdi-light-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/FileInputSteps.java]: fixed

**Pull request merged**